scylla-proxy/src/proxy.rs

```diff
@@ -61,12 +61,17 @@ impl ShardAwareness {
     }
 }
 
+/// Node can be either Real (truly backed by a Scylla node) or Simulated
```

Comment:
Nit: I think it would be actually fine to merge `proxy: Added Node::Simulated for dry mode` commit with the one that introduces the `NodeType::Simulated` enum variant - they are logically related to each other and one wouldn't obscure the other. Besides, you could avoid having to use `#[allow(unused)]`.